.github/workflows/npm-publish.yml

```
@@ -0,0 +1,31 @@
+# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
+
+name: 'Node.js Package'
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  publish-npm:
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: 'actions/checkout@v2'
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Cache Node dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - run: 'npm install'
+      - run: 'npm publish --access public'
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
```

```suggestion
      - name: Install dependencies
        run: 'npm install'
      
      - name: Run tests
         run: npm test
        
      - name: Publish to npm registry
         run: 'npm publish --access public'
         env:
           NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
```

I think it's a good idea to stop the publish action if the tests fails. That way you're less likely to publish a broken release.
I also added `name:` fields to describe each step.
I don't know why the formatting is off.

You should probably do a `npx prettier --write .github/workflows/npm-publish.yml` on this branch after committing my suggestions.